libs/vkd3d/command.c

```diff
@@ -907,8 +924,20 @@ static HRESULT STDMETHODCALLTYPE d3d12_fence_SetEventOnCompletion(d3d12_fence_if
 
     fence->events[fence->event_count].value = value;
     fence->events[fence->event_count].event = event;
+    fence->events[fence->event_count].latch = &latch;
     ++fence->event_count;
 
+    /* If event is NULL, we need to block until the fence value completes.
+     * Implement this in a uniform way where we pretend we have a dummy event.
+     * A NULL fence->events[].event means that we should set latch to true
+     * and signal a condition variable instead of calling external signal_event callback. */
+    if (!event)
+    {
+        latch = false;
+        while (!latch)
```

Comment:
do we not need atomic load/store for this, given that it happens on another thread?

Reply:
It happens while the fence object is locked, so not needed. It's a typical cond.wait([&]()); kind of thing.

Reply:
Signalling events/latches also happens while fence is locked.

Reply:
ah yeah, derp.